### PR TITLE
LPS-175873 Add Autom. Tests for FrontendDataSetAdminSearch

### PR DESCRIPTION
**Pull request by eltonccdantas**
Poshi Report for each test:

[FrontendDataSetAdminSearch#CanDisplayHeadlessDeliveryAPIResources](https://drive.google.com/file/d/1UcPxmliIh8_XCDUfdDavmhaPqNljgGlO/view?usp=share_link)
Test Plan
Given access Datasets admin page
When go to add Datasets
Then displays a list of headless resources

[FrontendDataSetAdminSearch#CanReturnAMatchedResult](https://drive.google.com/file/d/1JLQSrpDgWJrENK0_USwpAKon91QyiBho/view?usp=share_link)
Test Plan
Given access Datasets admin page
When go to add Datasets
And When search for a known uniquely named headless provider
Then the suggested list is updated to have only 1 result for the search
And Then the suggested list contains the uniquely named headless provider

[FrontendDataSetAdminSearch#CanReturnNoMatchedResults](https://drive.google.com/file/d/1gMWmiQd3H4-KOdGru1nDvSmKsegviDCD/view?usp=share_link)
Test Plan
Given access Datasets admin page
When go to add Datasets
And When search for a phrase that doesn't match available headless providers
Then suggested list contains 0 results

[Jira Ticket](https://issues.liferay.com/browse/LPS-175873)

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:help

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:test:frontend-data-set

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	@priority = 5
+	test CanDisplayHeadlessDeliveryAPIResources {
+		property custom.properties = "feature.flag.LPS-164563=true";
+
+		task ("Given access Datasets admin page") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Object",
+				panel = "Control Panel",
+				portlet = "Datasets");
+		}
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase:35]: This task is used in all tests, could you put it in the setup?

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 2]
```diff
@@ -241,7 +241,11 @@
 	<td>//label[contains(.,'${key_label}')]//following-sibling::div//button[contains(@class,'dropdown-toggle')]</td>
 	<td></td>
 </tr>
-
+<tr>
+	<td>EMPTY_DROPDOWN</td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:245]: Could you change the name of that path to something like DROPDOWN_ITEM?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+		Click(locator1 = "CreateObject#OBJECT_LABEL");
+
+		Type(
+			locator1 = "CreateObject#OBJECT_LABEL",
+			value1 = "${key_provider}");
+
+		KeyPress(
+			locator1 = "CreateObject#OBJECT_LABEL",
+			value1 = "\ENTER");
+	}
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro:13]: God job!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase (hunk 1 above)
Comment (mateusralv) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase:35]: set the FF at the beginning of the file, and remove it from tests.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase [hunk 4]
```diff
@@ -0,0 +1,120 @@
+@component-name = "portal-frontend-infrastructure"
+definition {
+
+	property osgi.modules.includes = "frontend-data-set-sample-web";
```
[the remark this answers is not in the document]
Comment (eltonccdantas) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase]: Done

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase (hunk 1 above)
Comment (eltonccdantas) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/frontenddataset/FrontendDataSetAdminSearch.testcase]: Done

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path (hunk 2 above)
Comment (eltonccdantas) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path]: Done

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro (hunk 3 above)
Comment (eltonccdantas) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSetAdmin.macro]: Done

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:frontend-data-set

**Comment by eltonccdantas** [comment]
ci:stop[frontend-data-set ]

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
continue: https://github.com/liferay-frontend/liferay-portal/pull/3089

**Comment by mateusralv** [comment]
Reopen because the failure is not related. Please see: https://liferay.slack.com/archives/C1SJ64S9X/p1677741205106169

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
Hi @eltonccdantas SF is green 🎉 🙌🏽

**Comment by mateusralv** [comment]
ci:test:frontend-data-set

**Review by mateusralv — APPROVED** [review]

**Comment by mateusralv** [comment]
ci:forward

**Pull request closed without merging**